storage/s3_storage.py

```diff
@@ -40,13 +40,19 @@ def save_to_filename(self, file_path: str) -> None:
         client = self._connect()
 
         transfer = boto3.s3.transfer.S3Transfer(client)
-        transfer.download_file(self._bucket, self._keyname, file_path)
+        try:
+            transfer.download_file(self._bucket, self._keyname, file_path)
+        except IOError:
+            raise NotFoundError("No Files Found")
```

Comment:
I don't know how much it matters, but you said `"No File Found"` in the Google implementation. Maybe only the `save_to_directory` and `delete_directory` functions need to use the plural form?